src/checkers/inference/util/defaults/InferenceQualifierDefaults.java

```diff
@@ -0,0 +1,47 @@
+package checkers.inference.util.defaults;
+
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.Element;
+import javax.lang.model.util.Elements;
+
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.util.defaults.Default;
+import org.checkerframework.framework.util.defaults.QualifierDefaults;
+
+import checkers.inference.qual.VarAnnot;
+import checkers.inference.util.ConstantToVariableAnnotator;
+
+public class InferenceQualifierDefaults extends QualifierDefaults {
+
+    public InferenceQualifierDefaults(Elements elements, AnnotatedTypeFactory atypeFactory) {
+        super(elements, atypeFactory);
+    }
+
+    @Override
+    protected DefaultApplierElement createDefaultApplierElement(AnnotatedTypeFactory atypeFactory,
+            Element annotationScope, AnnotatedTypeMirror type, boolean applyToTypeVar) {
+        return new InferenceDefaultApplierElement(atypeFactory, annotationScope, type, applyToTypeVar);
+    }
+
+    public class InferenceDefaultApplierElement extends DefaultApplierElement {
+
+        public InferenceDefaultApplierElement(AnnotatedTypeFactory atypeFactory, Element scope,
+                AnnotatedTypeMirror type, boolean applyToTypeVar) {
+            super(atypeFactory, scope, type, applyToTypeVar);
+        }
+
+        @Override
+        public void applyDefault(Default def) {
+            this.location = def.location;
```

Comment:
I'm wondering whether we could reduce duplication with super, by creating a changed `Default` object and then calling `super.applyDefault`. This might make the purpose of this method clearer. But it's just two lines, so might not be worth it.

Reply:
I think this might not worth the effort. The super class has a complicate logic of initializing sets of default qualifiers for checked code, unchecked code, and element default.

If we mutate the annotation stored in each `Default`, then we need to adapt the `checkDuplicate(DefaultSet, Annotation, TypeUseLocation)` method either... (That method has some logic on judging if the newly added `Annotation` is exisited or is a subtype of some qualifiers in the previous `DefaultSet`, the subtype judgement needs to adapt `qualifierHierarchy` so might need some amount of adaptation...)